rootex/framework/systems/light_system.cpp

```diff
@@ -56,13 +60,14 @@ LightsInfo LightSystem::getDynamicLights()
 		return Vector3::DistanceSquared(cameraPos, aa) < Vector3::DistanceSquared(cameraPos, bb);
 	};
 
-	Vector<PointLightComponent>& pointLightComponents = ECSFactory::GetAllPointLightComponent();
-	sort(pointLightComponents.begin(), pointLightComponents.end(), pointLightSortingLambda);
+	ComponentArray<PointLightComponent>& pointLightComponents = ECSFactory::GetAllPointLightComponent();
+	// TODO: Implement sorting. Currently due to raw pointers in components it is not possible to sort without guaranteeing there
```

Comment:
I think this can still be done. We don't actually need to sort the components in memory. We only need to find what is the order of the components in terms of distance from the camera.

Maybe just create an array of pointers to these components and then sort the array based on this criteria?

Reply:
Continued at https://github.com/sdslabs/Rootex/pull/589#discussion_r981163992